app/src/main/java/com/example/composestudy/presentation/SearchImageUI.kt

```diff
@@ -0,0 +1,191 @@
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.rememberAsyncImagePainter
+import com.example.composestudy.R
+import com.example.composestudy.domain.entity.Document
+import org.orbitmvi.orbit.compose.collectAsState
+
+@Composable
+fun SearchImageUI(viewModel: SearchViewModel = viewModel()) {
+    var searchText by remember { mutableStateOf(TextFieldValue()) }
+    val state by viewModel.collectAsState()
+
+    Surface(color = Color.White) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Top,
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            SearchBar(
+                searchText,
+                { query ->
+                    searchText = query
+                    viewModel.processAction(SearchAction.Search(query.text))
+                },
+                { searchText = TextFieldValue() },
+            )
+
+            when (val searchState = state.searchState) {
+                is SearchState.Loading -> {
+                    CircularProgressIndicator()
+                }
+
+                is SearchState.Success -> {
+                    SearchResults(searchState.results)
+                }
+
+                is SearchState.Error -> {
+                    val errorMessage = searchState.message
+                    Text(text = "Error: $errorMessage")
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun SearchBar(
+    searchText: TextFieldValue,
+    onSearchTextChanged: (TextFieldValue) -> Unit,
+    onClearSearch: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp, vertical = 16.dp)
+            .height(50.dp)
+            .background(Color.LightGray, shape = RoundedCornerShape(8.dp)),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.Start,
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.ic_search),
+            contentDescription = "Search Icon",
+            modifier = Modifier
```

Comment:
```suggestion
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 16.dp)
            .height(50.dp)
            .background(Color.LightGray, shape = RoundedCornerShape(8.dp)),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.Start,
    ) {
        Image(
            painter = painterResource(id = R.drawable.ic_search),
            contentDescription = "Search Icon",
            modifier = Modifier
```
위에 modifier 추가하면 Row 부모 노드에게만 modifier을 적용하고 나머지 자식 노드들은 Modifier 를 사용하게 됩니다!